### ISSUE TITLE
Feature: Allow sorting with commented areas

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [azat-io/eslint-plugin-perfectionist#22]
### What rule do you want to change?

sort-objects

### Describe the problem

Sometimes, there are properties that I would intentionally want to sort within specific delineated areas. For example, in https://github.com/JoshuaKGoldberg/template-typescript-node-package/pull/528/files#r1228571350, it was noted that I intentionally use comments to indicate areas of rules in my ESLint configs.

But, `perfectionist/sort-objects` is sorting them alphabetically as my lint config asked it to. It'd be nice to  have an option I can enable to sort _within_ commented areas.

~This feels like something that shouldn't be on by default. I think I'd only want it in specific files/areas, such as an ESLint config.~ Actually, I'm not sure... maybe this would be reasonable to turn on by default? Using `//` (non-JSDoc) comments to indicate specific areas is a reasonable practice...

### Code example

```js
{
    // First: these off-by-default rules are intentionally enabled
    "very-awesome-rule": "error",

    // Second: these on-by-default rules are intentionally disabled
    "not-as-awesome-rule": "off",
    "also-keep-in-this-area": "off",
}
```

After auto-fixing, that config gets turned into:

```js
export default {
    "also-keep-in-this-area": "off",

    // Second: these on-by-default rules are intentionally disabled
    "not-as-awesome-rule": "off",
    // First: these off-by-default rules are intentionally enabled
    "very-awesome-rule": "error",
}
```

### Additional comments

I checked the options in https://eslint-plugin-perfectionist.azat.io/rules/sort-objects#%F0%9F%94%A7-options and didn't see anything that looked related to this request.

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
Sounds like a good idea.

However, there are problems. Many people use such comments as comments to object properties. For example:

```
let user = {
  firstName: 'Kyle',
  lastName: 'Broflovski',
  // Get the full user name
  getFullName() {
      return [this.firstName, this.lastName].join(' ')
  },
  // Description to other function
  otherFunc() {
    /* ... */
  }
}
```

We have to take such comments into account when sorting.

1. What would the API for this functionality look like?
2. How can I figure out where the object's block partitioning comments are and where the property comments are?

**Comment by azat-io** [comment]
Alternatively, we can consider comments as separators if there is an empty line after them:

```js
{
    // First: these off-by-default rules are intentionally enabled

    "very-awesome-rule": "error",

    // Second: these on-by-default rules are intentionally disabled

    "not-as-awesome-rule": "off",
    "also-keep-in-this-area": "off",
}
```

What do you think about this idea?

**Comment by JoshuaKGoldberg** [comment]
> comments to object properties

Hmm yeah that's tricky. IMO property-specific comments should be done with proper `/** ... */` JSDoc comments, but a lot of devs don't actively use them...

> What would the API for this functionality look like?

🤔. Hmm. Maybe an option on the rule like `{ partitionByComment?: boolean }`? Or just `partition` for short (at expense of clarity)? `partitions`? `commentPartitions` 🤔 

> specific comments

Maybe `partition`/`partitionByComment` could be `boolean | RegExp`? So users could do specify matchers like `/^Sort:/`?

> if there is an empty line after them

Ehhhh I'm a little -1. That's a pretty particular coding style to go to. I personally don't use it because it separates the comments from what they're describing.

**Comment by azat-io** [comment]
I like the idea of adding the `partitionByComment` option. I will do it soon.

**Comment by azat-io** [comment]
Released in [v1.4.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v1.4.0)

[Docs](https://eslint-plugin-perfectionist.azat.io/rules/sort-objects#partition-by-comment).
